src/components/CardModal/CardModal.tsx

```diff
@@ -1,87 +1,73 @@
-import React, { useState } from 'react';
+import React from 'react';
 import './CardModal.scss';
-import { CardTypes } from '../../types/CardTypes';
-import { formatDate } from '../../helpers/functions/formatDate';
-import { useAppDispatch } from '../../app/hooks';
-import { updateCard } from '../../slices/columnSlice';
-import { updateTextAreaHeight } from '../../helpers/functions/updateTextAreaHeight';
-import { CardModalTitle } from '../CardModalTitle/CardModalTitle';
-import { CardModalDescription } from '../CardModalDescription/CardModalDescription';
-import { CardModalDate } from '../CardModalDate/CardModalDate';
+import { useForm } from 'react-hook-form';
+import { CardTypes } from 'types/CardTypes';
+import { formatDate } from 'helpers/functions/formatDate';
+import { useAppDispatch } from 'store/hooks';
+import { updateCard } from 'slices/columnSlice';
+import { yupResolver } from '@hookform/resolvers/yup';
+import descripLogo from 'img/description.svg';
+import { getCurrentColor } from 'helpers/functions/getCurrentColor';
+import { schema } from 'helpers/functions/schema';
 
 type Props = {
   card: CardTypes,
   onClose: () => void,
-  dueDateColor: string,
   onSaveName: () => void,
   editedName: string,
   onNameChange: (e: React.ChangeEvent<HTMLInputElement>) => void,
 };
 
+interface IFormInput {
+  name: string;
+  description?: string | null;
+  dueDate?: string | null;
+}
+
 export const CardModal: React.FC<Props> = ({
   card,
   onClose,
-  dueDateColor,
   onSaveName,
-  editedName,
-  onNameChange,
 }) => {
-  const [isDecriptionEdit, setIsDecriptionEdit] = useState(false);
-  const [isTitleEdit, setIsTitleEdit] = useState(false);
-  const { dueDate, description } = card;
-  const [editedDescription, setEditedDescription] = useState(description || '');
-  const [editedDueDate, setEditedDueDate] = useState(dueDate ? formatDate(dueDate) : '');
-  const [isDateInputEdit, setIsDateInputEdit] = useState(false);
-
+  const { dueDate, description, name } = card;
   const dispatch = useAppDispatch();
 
+  const { register, handleSubmit, formState: { errors }, } = useForm<IFormInput>({
+    defaultValues: {
+      name: name,
+      description: description,
+      dueDate: dueDate ? formatDate(dueDate) : '',
+    },
+    resolver: yupResolver(schema),
+  });
+
   const handleModalCloseClick = (e: React.MouseEvent) => {
     e.stopPropagation();
     onClose();
   };
 
-  const handleSave = (e: React.MouseEvent) => {
-    const updatedCard = {
-      ...card,
-      name: editedName,
-      description: editedDescription,
-    };
-
-    const newDueDate = new Date(editedDueDate); 
+  const onSubmit = (data: IFormInput) => {
+    let dueDateObject: Date | null = null;
```

Comment:
why variable have 'Object' in name if its not an object at all? Its not necessary to write type of variable in name, just name it as dueDate